docs/enhancements/0001-worker-pods.md

```diff
@@ -0,0 +1,223 @@
+This makes updating the DaemonSet spec impossible without interrupting the workload that depends on the kmod, because
+the DaemonSet controller will terminate the existing Pods and create new ones with the updated PodSpec.  
+Another issue is DaemonSet immutable fields; trying to update the `.spec.selector` will fail, which can make it
+challenging to improve the way KMM manages DaemonSets.
+
+### Resource utilization
+
+KMM currently creates one DaemonSet per `Module` and compatible kernel version found in the cluster.
+Formally, the number of DaemonSets created $S$ is defined by:
+$$S = \sum_{i=1}^nk_i$$
+
+where:
+- $n$ is the number of `Modules` in the cluster, and;
+- $k_i$ is the number of kernel versions running in the cluster that `Module` $i$ is compatible with.
+
+A big $S$ leads to unnecessary workload for the cluster:
+- on nodes, one ModuleLoader Pod is created per `Module`. Although those just sleep after the `modprobe`
+  `postStart` hook, each running pod incurs an overhead for the container runtime and the kubelet;
+- on the control-plane, a higher number of DaemonSets and Pods creates more events for kube-controller-manager, which
+  trigger more reconciliations and status updates.
+
+## Goals
+
+1. Improve the reliability of module loading and unloading
+2. Improve operator upgrade experience
+3. Reduce the overall KMM footprint
+
+## Non-goals
+
+1. Add further control on the deployment of kmods on nodes
+
+## Design
+
+### KMM: worker Pods pulling kmod images
+
+This proposal moves from ModuleLoader DaemonSets to standalone, one-shot Pods.
+KMM is now composed of two components:
+- the operator;
+- the worker Pods.
+
+```
+                                       ┌───────┐
+                                       │       │
+                                       │ Nodes ◄──┐
+                                       │       │  │
+                                       └───────┘  │ watches
+                                                  │
+                                     ┌─────────┐  │
+                                     │         │  │
+                                     │ Modules │  │     ┌───────────────────────┐
+                                     │         │  │     │                       │
+                                     └─────▲───┘  │     │ ┌───────────────────┐ │
+                                           │      │     │ │                   │ │    creates & owns
+                                watches &  └──────┴─────┼─┤ Module controller ├─┼─────────────────────┐
+                                updates status          │ │                   │ │                     │
+                                                        │ └───────────────────┘ │            ┌────────▼────────┐
+                                                        │                       │            │                 │
+                                                        │                       │            │ ModuleNodeState │
```

Comment:
What about renaming ModuleNodeState --> NodeModulesState? the structure is per Node and it may contain multiple Modules